extra_data/file_access.py

```diff
@@ -418,6 +427,10 @@ def index_groups(self, source):
             return {''}
         elif source in self.instrument_sources:
             return set(self.file[f'/INDEX/{source}'].keys())
+        elif source in self.reduction_data:
+            return set(self.file[f'/INDEX/{source}'].keys())
+        elif source in self.errata:
```

Comment:
I understand from our example that `REDUCTION` sources are always prepended by some category (e.g. `REDUCTION/PULSE_SLICING/<XTDF-source>/...`). How would `ERRATA` sources look like? All the reader machinery assumes that source names are unique across all root section.

Reply:
There's nothing concrete implemented in the DAQ, but some form of source identification is required. Think of a train that came outside the  buffer range, and couldn't be store. You'd want to know what the data was, so it would be prefixed by the source in the path.

Also technically all this additionally data is FAST data for the DAQ - i.e. it will be stored only if a train contains it.

Reply:
Certainly, my question was indeed more concrete in how you would pick the source name. Say `SA3_XTD10_XGM/XGM/DOOCS:output` has a hiccup and sends a late train, under what _source name_ (i.e. the patch below `ERRATA`) would you save this? It **must** be different from `SA3_XTD10_XGM/XGM/DOOCS:output` by at least a prefix or suffix.

Furthermore I had a look into your example file for data reduction. There are `INDEX` entries for
* PCLAYER_CI/DAQ/DET_DATA_TEST_1:xtdf
* PCLAYER_CI/LFF/DATA_TEST_1:reductionOutput
* PULSE_REDUCTION/PCLAYER_CI/DAQ/DET_DATA_TEST_1:xtdf

with a correspondig entry in `INSTRUMENT` for the first and in `REDUCTION` for the latter two. Is `PCLAYER_CI/LFF/DATA_TEST_1` here a real device used as input for the data aggregator, or also some virtual source used for metadata tracking? Ideally _sources_ in `REDUCTION` or `ERRATA` would **never** be identical to actual sources potentially occuring in `CONTROL` or `INSTRUMENT`.

Reply:
So in the reduction case, I prefix all source names with `PULSE_REDUCTION` to make them distinct from the "real" sources, one could do the same for `ERRATA`, e.g. `ERRATA/LATE_TRAINS/source/name/and/data/path/`. 

Hence, for what you observe in the index section:

* PCLAYER_CI/DAQ/DET_DATA_TEST_1:xtdf is the real source
* PCLAYER_CI/LFF/DATA_TEST_1:reductionOutput is the real source
* PULSE_REDUCTION/PCLAYER_CI/DAQ/DET_DATA_TEST_1:xtdf is injected by the pulse dim reducer, i.e. is an artificial source.